python/hail/linalg/matrix.py

```diff
@@ -47,27 +53,61 @@ def write(self, path):
     @handle_py4j
     @typecheck_method(that=block_matrix_type)
     def __mul__(self, that):
-        return BlockMatrix(self.hc,
-            self._jbm.multiply(that._jbm))
+        return BlockMatrix(self._jbm.multiply(that._jbm))
 
     @handle_py4j
     @typecheck_method(cols_to_keep=listof(integral))
     def filter_cols(self, cols_to_keep):
-        return BlockMatrix(self.hc,
-            self._jbm.filterCols(jarray(Env.jvm().long, cols_to_keep)))
+        return BlockMatrix(self._jbm.filterCols(jarray(Env.jvm().long, cols_to_keep)))
 
     @handle_py4j
     @typecheck_method(rows_to_keep=listof(integral))
     def filter_rows(self, rows_to_keep):
-        return BlockMatrix(self.hc,
-            self._jbm.filterRows(jarray(Env.jvm().long, rows_to_keep)))
+        return BlockMatrix(self._jbm.filterRows(jarray(Env.jvm().long, rows_to_keep)))
     
     @handle_py4j
     @typecheck_method(rows_to_keep=listof(integral),
                       cols_to_keep=listof(integral))
     def filter(self, rows_to_keep, cols_to_keep):
-        return BlockMatrix(self.hc,
-            self._jbm.filter(jarray(Env.jvm().long, rows_to_keep),
+        return BlockMatrix(self._jbm.filter(jarray(Env.jvm().long, rows_to_keep),
                              jarray(Env.jvm().long, cols_to_keep)))
 
+    @handle_py4j
+    def transpose(self):
+        return BlockMatrix(self._jbm.transpose())
+
+    def cache(self):
+        return BlockMatrix(self._jbm.cache())
+
+    @typecheck_method(storage_level=enumeration('NONE', 'DISK_ONLY', 'DISK_ONLY_2', 'MEMORY_ONLY',
```

Comment:
Nice.  I would define a typechecker for storage level and use this in VariantDataset/MatrixTable.persist, too.

Reply:
I think we do that in api2

Reply:
not explicitly. I changed them all. (not for api1, but for MatrixTable/Table.)